### PR TITLE
FTP Links By AccNo

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
Create an endpoint to generate FTP links by demand for a specific submission

### CODE REVIEW

**Comment by Juan-EBI** [comment]
changes looks good, but I think we need to include old files removal as part of links creation process

**Comment by jhoanmanuelms** [comment]
Done

File: Exporter/src/main/java/uk/ac/ebi/biostd/exporter/jobs/ftp/FtpService.java [hunk 1]
```diff
@@ -20,9 +21,35 @@
...
+    }
+
+    private void deleteOutdatedLinks(String submissionRelPath) {
+        File outdatedFtpLinks = new File(properties.getBaseFtpPath() + "/" +submissionRelPath);
+
+        if (outdatedFtpLinks.exists()) {
+            try {
+                FileUtils.deleteDirectory(outdatedFtpLinks);
+            } catch (IOException exception) {
+                log.info("Problem deleting outdated ftp links for {}", submissionRelPath);
```
Comment (Juan-EBI) [inline Exporter/src/main/java/uk/ac/ebi/biostd/exporter/jobs/ftp/FtpService.java]: probably we want here. Log.error and attach exception

File: Exporter/src/main/java/uk/ac/ebi/biostd/exporter/persistence/dao/SubmissionDao.java [hunk 2]
```diff
@@ -54,6 +54,11 @@ public void releaseSubmission(long submissionId) {
         return template.queryForList(queries.getPublicSubmissions(), emptyMap(), String.class);
     }
 
+    public String getPublicSubmissionPathByAccNo(String accNo) {
```
Comment (Juan-EBI) [inline Exporter/src/main/java/uk/ac/ebi/biostd/exporter/persistence/dao/SubmissionDao.java]: `getSubmissionPathByAccNo` as the operation is not related to submission visibility.

File: Exporter/src/test/java/uk/ac/ebi/biostd/exporter/test/FtpServiceTest.java [hunk 3]
```diff
@@ -57,16 +59,46 @@ public void generateFtpLinks() {
...
+        assertFalse(Files.exists(Paths.get(
+            folder.getRoot().getAbsolutePath() + "/ftp/S-EPMC/S-EPMCxxx633/S-EPMC3343633/OutdatedFile1.txt")));
+    }
+
+    @Test
+    public void generateFtpLinksByAccNo() {
+        testInstance.execute("S-EPMC3343634");
+
+        assertFalse(Files.exists(Paths.get(
+            folder.getRoot().getAbsolutePath() + "/ftp/S-EPMC/S-EPMCxxx633/S-EPMC3343633/File1.txt")));
```
Comment (Juan-EBI) [inline Exporter/src/test/java/uk/ac/ebi/biostd/exporter/test/FtpServiceTest.java]: The looks kind of verbose not  we may create utility methods as  `assertFtpFileExist("S-EPMC3343633/File1.txt")`

File: Exporter/src/test/java/uk/ac/ebi/biostd/exporter/test/FtpServiceTest.java [hunk 4]
```diff
@@ -57,16 +59,46 @@ public void generateFtpLinks() {
...
         folder.newFile("submissions/S-EPMC/S-EPMCxxx633/S-EPMC3343633/Files/File1.txt");
+        folder.newFile("submissions/S-EPMC/S-EPMCxxx634/S-EPMC3343634/Files/File2.txt");
+    }
+
+    private void createOutdatedLinks() throws IOException {
+        folder.newFolder("ftp");
+
+        folder.newFolder("ftp", "S-EPMC");
+        folder.newFolder("ftp", "S-EPMC", "S-EPMCxxx633");
+        folder.newFolder("ftp", "S-EPMC", "S-EPMCxxx633", "S-EPMC3343633");
```
Comment (Juan-EBI) [inline Exporter/src/test/java/uk/ac/ebi/biostd/exporter/test/FtpServiceTest.java]: is not this equivalente to only
```
 folder.newFolder("ftp", "S-EPMC", "S-EPMCxxx633", "S-EPMC3343633");
```

**Pull request merged**